src/main/java/com/techsophy/tsf/runtime/form/dto/Validate.java

```diff
@@ -0,0 +1,130 @@
+package com.techsophy.tsf.runtime.form.dto;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.techsophy.tsf.runtime.form.config.GlobalMessageSource;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+import static com.techsophy.tsf.runtime.form.constants.ErrorConstants.*;
+import static com.techsophy.tsf.runtime.form.constants.FormModelerConstants.*;
+
+@Data
+@AllArgsConstructor(onConstructor_ = {@Autowired})
+public class Validate
+{
+   Boolean required;
+   String pattern;
+   String customMessage;
+   String custom;
+   Boolean customPrivate;
+   String json;
+   Double min;
+   Double max;
+   Integer minLength;
+   Integer maxLength;
+   Boolean strictDateValidation;
+   Boolean multiple;
+   Boolean unique;
+   Integer minWords;
+   Integer maxWords;
+   String step;
+   String integer;
+
+
+   public List<ValidationResult> validate(ComponentData compData,GlobalMessageSource globalMessageSource,MongoTemplate mongoTemplate)
+   {
+      List<ValidationResult> validationResultList=new ArrayList<>();
+      Component component=compData.comp;
```

Comment:
make component and data private, and only expose helper methods to retrieve specific details.